src/ruby/puppet-server-lib/puppet/server/master.rb

```diff
@@ -97,7 +97,7 @@ def getClassInfoForEnvironment(env)
     # and so would not be subject to any potentially stale settings data being
     # used to enumerate manifests.
     environment = @env_loader.list.find do |env_from_loader|
-      env_from_loader.name == env
+      env_from_loader.name.to_s == env
```

Comment:
I can't really explain why this change is needed now but these tests were able to pass without it previously.  This allows the comparison to be made as a String to a String rather than as a Symbol to a String.  Locally, I haven't seen Symbol to String comparisons succeed in other contexts and it doesn't appear that the type of `Puppet::Node::Environment.name` has changed from String to Symbol, at least recently.